src/zkevm_specs/evm/execution/extcodehash.py

```diff
@@ -1,37 +1,38 @@
+from ...util import EMPTY_CODE_HASH, EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS, FQ, N_BYTES_ACCOUNT_ADDRESS
 from ..instruction import Instruction, Transition
-from ..table import CallContextFieldTag, AccountFieldTag
-from ...util.param import EXTRA_GAS_COST_ACCOUNT_COLD_ACCESS
-from ...util.hash import EMPTY_CODE_HASH
-from ...util import FQ
+from ..opcode import Opcode
+from ..table import AccountFieldTag, CallContextFieldTag
 
 
 def extcodehash(instruction: Instruction):
     opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.EXTCODEHASH)
 
-    address = instruction.rlc_to_fq(instruction.stack_pop(), 20)
+    address = instruction.rlc_to_fq(instruction.stack_pop(), N_BYTES_ACCOUNT_ADDRESS)
 
     tx_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
     is_warm = instruction.add_account_to_access_list(tx_id, address, instruction.reversion_info())
 
-    nonce = instruction.account_read(address, AccountFieldTag.Nonce)
-    balance = instruction.account_read(address, AccountFieldTag.Balance)
-    code_hash = instruction.account_read(address, AccountFieldTag.CodeHash)
+    # Load account `exists` value from auxilary witness data.
+    exists = instruction.curr.aux_data
 
-    is_empty = (
-        instruction.is_zero(nonce)
-        * instruction.is_zero(balance)
-        * instruction.is_equal(code_hash, instruction.rlc_encode(EMPTY_CODE_HASH, 32))
-    )
+    if exists == 0:  # 1 - exists == 1
+        instruction.account_read(address, AccountFieldTag.NonExisting)
 
+    code_hash = (
+        instruction.account_read(address, AccountFieldTag.CodeHash)
+        if exists == 1
+        else instruction.rlc_encode(EMPTY_CODE_HASH, 32)
```

Comment:
maybe this `else` is unneeded: when `exists == 0`, the `code_hash` value will not be used.

Reply:
Fixed to only set `code_hash` when `exists == 1` as:
```
if exists == 1:
    code_hash = instruction.account_read(address, AccountFieldTag.CodeHash)
else:  # exists == 0
    instruction.account_read(address, AccountFieldTag.NonExisting)
```
And also updated code in `balance.py`.

Reply:
I am sorry that I missed `make install` to test this code 😂. Cell `code_hash` must be assigned. I reverted the code as:
```
if exists == 0:
    instruction.account_read(address, AccountFieldTag.NonExisting)

code_hash = (
    instruction.account_read(address, AccountFieldTag.CodeHash) if exists == 1 else RLC(0)
)
```
Similar as `BALANCE`.

Reply:
ah no problem! This looks good :)